jenkins/hourly_maintenance.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import collections
+import datetime
+import re
+import subprocess
+import sys
+
+
+def ContainerImages():
+  for line in subprocess.check_output([
+      'docker',
+      'ps', '-a',
+      '--format={{.Image}}']).split('\n'):
+    if not line:
+      continue
+    yield line
+
+
+def RemoveContainers():
+  """Remove non-running containers that we created a long time ago."""
+  now = datetime.datetime.now()
+  old = []
+  for line in subprocess.check_output([
+      'docker',
+      'ps', '-a',
+      '-f', 'status=created',  # Never started due to timeout
+      '-f', 'status=exited',  # Container exited
+      '-f', 'status=dead',  # Zombie container
+      '--format={{.CreatedAt}}\t{{.ID}}\t{{.Image}}',
+  ]).split('\n'):
```

Comment:
Can you guarantee that this is run in a UTC environment? I'm a little worried about the 2 hour limit interacting with a PDT timezone and experiencing skew. You can override the TZ in the subprocess invocation.